base/src/commonTest/kotlin/state/ColdStateRepoTest.kt

```diff
@@ -64,12 +77,15 @@ class ColdStateRepoTest : BaseTest() {
 
     private suspend fun Repo.testCollectionIsCalledTimes(times: Int) {
         val job = launch { collect() }
+        // For some reason yield is not reliable here
```

Comment:
the suspend context here is the runBlocking from the test (on a background thread), the repo has its own scope combined with a Main dispatcher. So in essence there is nothing to yield to, we just need to wait for the thread context switch to happen on the OS level.

passing the context into the` Repo()` might improve the test in that regard.